### PR TITLE
Fix host on darwin

### PR DESCRIPTION
**Pull request by peterbourgon**
Let's see if Circle will work...

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -37,7 +37,8 @@ test:
     - cd $SRCDIR; ./bin/lint .
     - cd $SRCDIR; make client-test
     - cd $SRCDIR; make static
-    - cd $SRCDIR; make
+    - cd $SRCDIR; rm -f app/app probe/probe; GOOS=linux make
+    - cd $SRCDIR; rm -f app/app probe/probe; GOOS=darwin make
```
Comment (tomwilkie) [inline circle.yml]: Is this going to build a darwin executable (and stick it in the docker container, and upload it to the docker hub)?

Comment (peterbourgon) [inline circle.yml]: Ha, good catch.

File: probe/host/system_test.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package host_test
```
Comment (tomwilkie) [inline probe/host/system_test.go]: These tests are a bit weak; why did you remove the original unit tests (that mock syscall.Uname and Readfile?)

Comment (peterbourgon) [inline probe/host/system_test.go]: - I didn't remove any tests, only added some
- You can't mock syscall.Uname, as it's only defined for GOOS=linux
- You can't really mock ReadFile, as the files you're reading (/proc) don't exist on Darwin, so the implementation would fail...

Comment (tomwilkie) [inline probe/host/system_test.go]: You can have platform specific tests...

Comment (peterbourgon) [inline probe/host/system_test.go]: Of course, but then you're just mocking and testing for a single platform. It seems like it makes more sense to move the mocking boundary "up" one level, and mock the functions that actually return cross-platform-usable data. Incidentally, that's why I added system_test.go — to make sure the underlying implementations return sane data.

**Comment by tomwilkie** [comment]
LGTM with squash
